### PR TITLE
added hibernate-criteria-query-homework solution

### PR DESCRIPTION
**Pull request by Basilius-Third**

### CODE REVIEW

File: src/main/java/ma/hibernate/dao/PhoneDaoImpl.java [hunk 1]
```diff
@@ -12,11 +18,44 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
...
+                CriteriaBuilder.In<String> predicate = cb.in(phoneRoot.get(entry.getKey()));
+                for (String value : entry.getValue()) {
+                    predicate.value(value);
+                }
+                largePredicate = cb.and(largePredicate, predicate);
+            }
+            query.where(largePredicate);
+            return session.createQuery(query).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find all by given params: " + params, e);
```
Comment (PashaTerletskyi) [inline src/main/java/ma/hibernate/dao/PhoneDaoImpl.java:58]:
```suggestion
            throw new RuntimeException("Can't find all by given params", e);
```

**Review by PashaTerletskyi — COMMENTED** [review]
Good job!)

**Review by PashaTerletskyi — APPROVED** [review]

File: src/main/java/ma/hibernate/dao/PhoneDaoImpl.java [hunk 2]
```diff
@@ -12,11 +18,44 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
...
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
```
Comment (KseniiaMakarova) [inline src/main/java/ma/hibernate/dao/PhoneDaoImpl.java:44]: is this naming ok?

File: src/main/java/ma/hibernate/dao/PhoneDaoImpl.java [hunk 3]
```diff
@@ -12,11 +18,44 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
...
+                CriteriaBuilder.In<String> predicate = cb.in(phoneRoot.get(entry.getKey()));
+                for (String value : entry.getValue()) {
+                    predicate.value(value);
+                }
+                largePredicate = cb.and(largePredicate, predicate);
+            }
+            query.where(largePredicate);
+            return session.createQuery(query).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find all by given params", e);
```
Comment (KseniiaMakarova) [inline src/main/java/ma/hibernate/dao/PhoneDaoImpl.java:58]:
```suggestion
            throw new RuntimeException("Can't find all phones by given params", e);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**